components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushPatientDataTest.java

```diff
@@ -0,0 +1,154 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.phenotips.data.push.PushPatientData;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.context.Execution;
+import org.xwiki.context.ExecutionContext;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+public class DefaultPushPatientDataTest {
+
+    @Rule
+    public final MockitoComponentMockingRule<PushPatientData> mocker =
+            new MockitoComponentMockingRule<PushPatientData>(DefaultPushPatientData.class);
+
+    @Mock
+    private BaseObject serverConfiguration;
+
+    @Mock
+    private List<NameValuePair> data;
+
+    @Mock
+    private XWiki xWiki;
+
+    @Mock
+    private XWikiContext xWikiContext;
+
+    @Mock
+    private XWikiDocument xWikiDocument;
+
+    @Mock
+    private ExecutionContext executionContext;
+
+    private Execution execution;
+
+    private DefaultPushPatientData patientData;
+
+    @Before
+    public void setUp() throws ComponentLookupException
+    {
+        MockitoAnnotations.initMocks(this);
+        this.patientData = (DefaultPushPatientData) this.mocker.getComponentUnderTest();
+        this.execution = this.mocker.getInstance(Execution.class);
+    }
+
+    @Test
+    public void getBaseUrlReturnsCorrectString() throws ComponentLookupException, NoSuchMethodException, IllegalAccessException,
+            InvocationTargetException {
+        // set up access for the private method
+        Method method = DefaultPushPatientData.class.getDeclaredMethod("getBaseURL", BaseObject.class);
+        method.setAccessible(true);
+        // test null case
+        Assert.assertNull(method.invoke(this.mocker.getComponentUnderTest(), serverConfiguration));
+
+        // test case when serverConfiguration gives http string
+        when(serverConfiguration.getStringValue("url")).thenReturn("http://");
+        Assert.assertEquals(method.invoke(this.mocker.getComponentUnderTest(), serverConfiguration),
+                "http:/bin/receivePatientData");
+
+        when(serverConfiguration.getStringValue(("url"))).thenReturn("apples");
+        Assert.assertEquals(method.invoke(this.mocker.getComponentUnderTest(), serverConfiguration),
+                "http://apples/bin/receivePatientData");
+    }
+
+    @Test
+    public void generateRequestDataTest() throws ComponentLookupException, NoSuchMethodException, IllegalAccessException,
+            InvocationTargetException {
+        // set up access for the private method
+        Method method = DefaultPushPatientData.class.getDeclaredMethod("generateRequestData",
```

Comment:
No, don't do that. Private methods shouldn't be tested, you should stick to the methods exposed by the interface. Some other method should be calling the private method, so you must test that one instead.